### PR TITLE
KOGITO-8147: Refactor the app bar and menu according to UX redesign

### PR DESCRIPTION
**Pull request by bxf12315**

### CODE REVIEW

**Comment by domhanak** [comment]
@tomasdavidorg this will most probably influence the it-tests, tagging you just in case

**Comment by tomasdavidorg** [comment]
@bxf12315 Is it still draft or is it ready for review?

**Comment by paulovmr** [comment]
Hi @tomasdavidorg , it is still a draft at this time. We actually have a few redesign tasks to come, but there will be a separate upstream branch where they will be merged first before going to main, so we can verify them and have some time to change UI tests first. This PR will be changed to that branch instead of main.

**Comment by tomasdavidorg** [comment]
@paulovmr Thanks.

File: packages/serverless-logic-sandbox/package.json [hunk 1]
```diff
@@ -19,6 +19,8 @@
     "build:prod": "pnpm lint && pnpm test && rimraf dist && webpack && pnpm test:it",
     "cy:open": "cypress open --project it-tests --config baseUrl=$(build-env serverlessLogicSandbox.dev.cypressUrl)",
     "cy:run": "cypress run --headed -b chrome --project it-tests --config baseUrl=$(build-env serverlessLogicSandbox.dev.cypressUrl)",
+    "format": "prettier --write  './src/**/*.{tsx,ts,js}'",
+    "postinstall": "patch-package",
```
Comment (paulovmr) [inline packages/serverless-logic-sandbox/package.json]: Why was this added here?

File: packages/serverless-logic-sandbox/src/navigation/RoutesSwitch.tsx [hunk 2]
```diff
@@ -53,7 +53,7 @@ export function RoutesSwitch() {
           />
         )}
       </Route>
-      <Route exact={true} path={routes.home.path({})}>
```
Comment (paulovmr) [inline packages/serverless-logic-sandbox/src/navigation/RoutesSwitch.tsx:56]: Why was the `exact` removed?

File: packages/serverless-logic-sandbox/src/newHomepage/breadcrumb/HomepageBreadcrumb.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * limitations under the License.
+ */
+
+import * as React from "react";
+import { Breadcrumb, BreadcrumbItem } from "@patternfly/react-core";
+
+interface IOwnProps {
+  paths: string[];
+}
+const HomepageBreadcrumb: React.FC<IOwnProps> = (props) => {
```
Comment (paulovmr) [inline packages/serverless-logic-sandbox/src/newHomepage/breadcrumb/HomepageBreadcrumb.tsx]:
```suggestion
export function HomepageBreadcrumb(props: { paths: string[] }) => {
```
As a convention, we use functions when defining new components in this repository. Please do the same for the other components.

File: packages/serverless-logic-sandbox/src/newHomepage/HomePage.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ */
+
+import * as React from "react";
+import { useMemo } from "react";
+import { HashRouter, Route } from "react-router-dom";
+import OnlineEditorPage from "./pageTemplate/OnlineEditorPage";
+import { Label } from "@patternfly/react-core/dist/js/components/Label";
+import HomePageRoutes from "./routes/HomePageRoutes";
+
+const HomePage: React.FC = () => {
```
Comment (paulovmr) [inline packages/serverless-logic-sandbox/src/newHomepage/HomePage.tsx]: There is already a HashRouter in the `RoutesSwitch.tsx` file, so this file should not exist, we can use that Router to include the new pages.

File: packages/serverless-logic-sandbox/src/newHomepage/breadcrumb/HomepageBreadcrumb.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    <Breadcrumb>
+      {props.paths.map((path) => (
+        <BreadcrumbItem to={getPath(path)}>{path === "/" ? { path } : "Home"}</BreadcrumbItem>
+      ))}
+      ;
+    </Breadcrumb>
+  );
+};
+
+export default HomepageBreadcrumb;
```
Comment (paulovmr) [inline packages/serverless-logic-sandbox/src/newHomepage/breadcrumb/HomepageBreadcrumb.tsx]:
```suggestion
}
```
Following the commentary above.

File: packages/serverless-logic-sandbox/src/newHomepage/uiNav/HomePageNav.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        <NavItem itemId={1} key={"Serverless-models-nav"} isActive={pathname === "/ServerlessModels"}>
+          <Link to="/ServerlessModels">Serverless Models</Link>
+        </NavItem>
+
+        <NavItem itemId={2} key={"SampleCatalog-nav"} isActive={pathname === "/SampleCatalog"}>
+          <Link to="/SampleCatalog">Sample Catalog</Link>
+        </NavItem>
+
+        <NavItem itemId={3} key={"Documentation-nav"} isActive={pathname === "/Documentation"}>
+          <Link to="/Documentation">Documentation</Link>
```
Comment (paulovmr) [inline packages/serverless-logic-sandbox/src/newHomepage/uiNav/HomePageNav.tsx:38]: This link must open a new tab.

**Review by paulovmr — CHANGES_REQUESTED** [review]
Hi @bxf12315 , I did a partial review with several suggestions. Besides that, I see that the layout in the new pages deserve a few improvements, like padding in some instances.

In my opinion, in this task we shouldn't have the new homepage in parallel with the current one. Please replace the files when appropriate, and instead of sending this PR to `main`, I created a branch `KOGITO-8015-feature-preview` in this repo to concentrate the redesign PRs. I will send another review tomorrow about other files.

File: packages/serverless-logic-sandbox/src/newHomepage/HomePage.tsx (hunk 4 above)
Comment (bxf12315) [reply packages/serverless-logic-sandbox/src/newHomepage/HomePage.tsx]: I do not understand, I think we need to use App.tsx.

**Comment by bxf12315** [comment]
I will close this one and reopen it at https://github.com/kiegroup/kie-tools/pull/1335
